components/identity/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -79,7 +80,10 @@ public OAuth2AuthorizeRespDTO authorize(OAuth2AuthorizeReqDTO oAuth2AuthorizeReq
         try {
             AuthorizationHandlerManager authzHandlerManager =
                     AuthorizationHandlerManager.getInstance();
-            return authzHandlerManager.handleAuthorization(oAuth2AuthorizeReqDTO);
+            OAuth2AuthorizeRespDTO oAuth2AuthorizeRespDTO =
+                    authzHandlerManager.handleAuthorization(oAuth2AuthorizeReqDTO);
+            oAuth2AuthorizeRespDTO.setTokenType(BEARER_TOKEN_TYPE);

```

Comment:
This needs to be done inside the issue() method of the AuthorizationHandlers. The AuthorizationHandlers are extensible andiIf you fix it here regardless of what the handlers set you will be overwriting it here.